### PR TITLE
Add options to URLLauncher to use WebView and SafariVC

### PR DESCRIPTION
**Pull request by mehmetf**
This will fix https://github.com/flutter/flutter/issues/13059.

### CODE REVIEW

**Comment by xster** [comment]
Should we just do a open in app or launch new task toggle for both android and ios?

**Comment by mehmetf** [comment]
@xster Please clarify with an example? Are you referring to supporting this on Android via a WebView perhaps?

File: packages/url_launcher/ios/Classes/UrlLauncherPlugin.m [hunk 1]
```diff
@@ -67,11 +67,15 @@ - (instancetype)initWithViewController:(UIViewController *)viewController {
...
-  NSString *url = call.arguments;
+  NSString *url = call.arguments[@"url"];
   if ([@"canLaunch" isEqualToString:call.method]) {
     result(@([self canLaunchURL:url]));
   } else if ([@"launch" isEqualToString:call.method]) {
-    [self launchURL:url result:result];
+    if (call.arguments[@"useSafariVC"]) {
+      [self launchURLInVC:url result:result];
+    } else {
+      [self launchURLInBrowser:url result:result];
```
Comment (mit-mit) [inline packages/url_launcher/ios/Classes/UrlLauncherPlugin.m]: The URL might something else than HTML (mails to, and), so maybe this name is too web specific? handleInRegistredApp?

Comment (mehmetf) [reply packages/url_launcher/ios/Classes/UrlLauncherPlugin.m]: Agreed. Changed to "launchURL".

File: packages/url_launcher/README.md [hunk 2]
```diff
@@ -67,3 +67,9 @@ best practice we suggest adjusting the application UI so that the unsupported
 URL is never triggered; for example, if the `email` scheme is not supported, a
 UI button that would have sent email can be changed to redirect the user to a
 web page using a URL following the `http` scheme.
+
+## Browser vs In-app Handling
+By default, Android opens up a browser when handling URLs. You can pass
+forceWebView: true parameter to tell the plugin to open a WebView instead. On
+iOS, the default behavior is to open all web URLs within the app. Everything
+else is redirected to the app handler.
```
Comment (xster) [inline packages/url_launcher/README.md]: Document the forceSafariVC too for the cookie sharing use case?

File: packages/url_launcher/lib/url_launcher.dart [hunk 3]
```diff
@@ -15,10 +15,38 @@ const MethodChannel _channel =
 /// The returned future completes with a [PlatformException] on invalid URLs and
 /// schemes which cannot be handled, that is when [canLaunch] would complete
 /// with false.
-Future<Null> launch(String urlString) {
+///
+/// [forceSafariVC] is only used in iOS. If unset, the launcher opens web URLs
+/// in the safari VC, anything else is opened using the default handler on the
+/// platform. If set to true, it opens the URL in the Safari view controller.
```
Comment (xster) [inline packages/url_launcher/lib/url_launcher.dart]: Describe failure mode if not openable in safari?

File: packages/url_launcher/README.md (hunk 2 above)
Comment (mehmetf) [reply packages/url_launcher/README.md]: Webviews can and do share the browser context. Mentioned it in the doc.

File: packages/url_launcher/lib/url_launcher.dart (hunk 3 above)
Comment (mehmetf) [reply packages/url_launcher/lib/url_launcher.dart]: Done.

**Comment by mit-mit** [comment]
fyi @mehmetf I have pushed this to pub

**Pull request merged**